### PR TITLE
feat: added cart & wishlist management

### PR DESCRIPTION
**Pull request by kedark152**
### feat: added Cart & Wishlist Management
**Preview Link: https://deploy-preview-3--agro-stores.netlify.app/**

**Features added in this Pull Request are :**

**Cart Management**
- In Product Listing Page we can click on **'Add to cart'** for adding products in Cart.
- After the product is added to cart, the same button will change to **'Go to Cart'**
- Also from navbar, we can navigate to the cart page where all the products we want to buy are mentioned.
- In My Cart Page We can see the Price Details Summary , Delivery Charge Discount applied in cart page. 
- We can update the quantity of items so the price will also get updated in Price Details Summary.
- We can remove the product from the cart by clicking "remove".

**Wishlist Management**
- In Product Listing Page we can click on **'Heart Icon'** for adding products in Wishlist.
- From navbar, we can navigate to the Wishlist page where all the products we have added in Wishlist are mentioned.
- We can remove the product from Wishlist by clicking on 'Heart-Icon'.
- We can also add/remove products in Wishlist from My Cart page.

### CODE REVIEW

File: src/components/CartPriceCard.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+function totalOldPrice(){
+  let collectOldPrice = 0;
+   cartState.cartItems.map((item)=>collectOldPrice = collectOldPrice+(item.price_old*item.quantity));
+   return collectOldPrice;
+  }
+
+  const deliveryCharge = () =>(totalPrice()>499?0:100);
+     
+  
+
```
Comment (junaidshaikh-js) [inline src/components/CartPriceCard.js]: Too much extra spaces is added

**Review by junaidshaikh-js — APPROVED** [review]
Good work! 

Clean code, liked the idea of separating the update function which makes the code more cleaner. Can be merged.

**Review by Bharatrathod-81 — APPROVED** [review]
looks perfect

File: src/components/CartPriceCard.js (hunk 1 above)
Comment (kedark152) [reply src/components/CartPriceCard.js]: updated, Thanks for doing PR Review

**Pull request merged**